### PR TITLE
Add more details for constants

### PR DESCRIPTION
**Pull request by tokesk**
- [ ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `master` branch of freeCodeCamp.
- [ ] None of my changes are plagiarized from another source without proper attribution.
- [ ] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

File: guide/english/php/constants/index.md [hunk 1]
```diff
@@ -2,11 +2,35 @@
...
+
+The `define()` function to set a constant takes three arguments - the key name, the key's value, and a Boolean (true or false) which determines whether the key's name is case-insensitive (false by default). A constant's value cannot be altered once it is set. It is used for values which rarely change (for example a database password OR api key).
+
+The name of a constant follows the same rules as any label in PHP. A valid constant name starts with a letter or underscore, followed by any number of letters, numbers, or underscores. As a regular expression, it would be expressed thusly: *[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]* *
+
 
 ### Scope
 It is important to know that unlike variables, constants ALWAYS have a global scope and can be accessed from any function in the script.
 
+### Example - Valid and invalid constant names
```
Comment (co16353sidak) [inline guide/english/php/constants/index.md]: In adding this example, you have in the process created a duplicate Example heading which shouldn't be there,
make it such that headings should be in the format `Example # - Description` with # being the number of the example and description being a small description describing the example.

File: guide/english/php/constants/index.md [hunk 2]
```diff
@@ -2,11 +2,35 @@
...
+
+// Invalid constant names
+define("2FOO",    "something");
+
+// This is valid, but should be avoided:
+// PHP may one day provide a magical constant
+// that will break your script
+define("__FOO__", "something"); 
+```
+
```
Comment (co16353sidak) [inline guide/english/php/constants/index.md]: As a practice it is a suggestion to add output alongside the code to allow for the reader to understand how constants will work, follow the format in the below example.

**Review by theNaut1lus — CHANGES_REQUESTED** [review]
I have left 2 suggestive comments, please review on them both and make the required changes.

**Comment by RandellDawson** [comment]
Hi @tokesk ,

As there has been no response from your end for a while, we are closing this PR.

Feel free to request this PR to be reopened or create a new PR, if you still wish to make the changes you originally added plus any additional change requests asked of you.  Also, remember to update your fork, so you have the latest version of master in case it has changed significantly.

Happy Coding!

**Pull request closed without merging**